ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -1010,18 +1034,24 @@ public static boolean setUserMetadataHeaders(byte[] userMetadata, RestResponseCh
    * Sets the user metadata in the headers of the response.
    * @param userMetadataMap map of user metadata that needs to be sent.
    * @param restResponseChannel the {@link RestResponseChannel} that is used for sending the response.
+   * @param keysToNotPrefix a set of user metadata keys to not add user metadata prefix in the response header
    * @throws RestServiceException if there are any problems setting the header.
    */
   public static void setUserMetadataHeaders(Map<String, String> userMetadataMap,
-      RestResponseChannel restResponseChannel) {
+      RestResponseChannel restResponseChannel, Set<String> keysToNotPrefix) {
     Objects.requireNonNull(userMetadataMap, "user metadata must be supplied");
+    Objects.requireNonNull(keysToNotPrefix, "keys to not prefix must be supplied");
     for (Map.Entry<String, String> entry : userMetadataMap.entrySet()) {
       String headerName = entry.getKey();
       String headerValue = entry.getValue();
       // Ensure um prefix is prepended to header name
       if (!headerName.startsWith(Headers.USER_META_DATA_HEADER_PREFIX)) {
         headerName = Headers.USER_META_DATA_HEADER_PREFIX + headerName;
       }
+      String headerNameWithoutPrefix = headerName.substring(Headers.USER_META_DATA_HEADER_PREFIX.length());
+      if (keysToNotPrefix.contains(headerNameWithoutPrefix)) {
+        headerName = headerNameWithoutPrefix;
```

Comment:
Is there any possibility that after removing the prefix, the headName is same as some Ambry using standard header?
Also also does that mean customer can expose any headerName as they want with the container_without_prefix? Or probably we have some constraint for the container_without_prefix?

Reply:
That's a solid concern, however, it's probably fine. Since what goes to this set is controlled us in the end. We will manually examine the keys and make sure it doesn't conflict with standard http response header keys.

Reply:
Yes, as long as we exam the list well in the container setting, it should be fine.